src-electron/zcl/zcl-loader-silabs.js

```diff
@@ -1060,7 +1060,8 @@ async function processDataType(
 function prepareNumber(a, dataType) {
   return {
     size: a.$.size,
-    is_signed: a.$.name.endsWith('u') || !a.$.name.includes('int') ? 0 : 1,
+    is_signed:
```

Comment:
How does this handle signed types like `temp-diff` or `temperature`, which are named integer types derived from signed types?

Reply:
The loader first loads atomic types and then derived ones but we may not be doing that for numbers. Could you point me to an existing derived type for a number? I can test that. Any reason we have to define a type for temperature and not just use one of the existing number type for it?

Reply:
@brdandu I'm not sure `temperature` and `temp-diff` are used in the Matter XML right now, but the Matter spec does define them in the Thermostat cluster.  So ideally the XML would use them, to match the spec.

The reason to use a named type is that the named type can define what values actually mean.

Ideally, it seems to me, `<type>` would have a signed/unsigned marker thing when `analog="true"`....  

Anyway derived types for number would be all of these, in Matter:
```
    <type id="0xE0" description="Time of day"              name="tod"               size="4"  analog="true"   />
    <type id="0xE1" description="Date"                     name="date"              size="4"  analog="true"   />
    <type id="0xE2" description="UTC Time"                 name="utc"               size="4"  analog="true"   />
    <type id="0xE3" description="Epoch Microseconds"       name="epoch_us"          size="8"  analog="true"   />
    <type id="0xE4" description="Epoch Seconds"            name="epoch_s"           size="4"  analog="true"   />
    <type id="0xE5" description="System Time Microseconds" name="systime_us"        size="8"  analog="true"   />
    <type id="0xE6" description="Percentage units 1%"      name="percent"           size="1"  analog="true"   />
    <type id="0xE7" description="Percentage units 0.01%"   name="percent100ths"     size="2"  analog="true"   />
    <type id="0xE8" description="Cluster ID"               name="cluster_id"        size="4"  discrete="true" />
    <type id="0xE9" description="Attribute ID"             name="attrib_id"         size="4"  discrete="true" />
    <type id="0xEA" description="Field ID"                 name="field_id"          size="4"  discrete="true" />
    <type id="0xEB" description="Event ID"                 name="event_id"          size="4"  discrete="true" />
    <type id="0xEC" description="Command ID"               name="command_id"        size="4"  discrete="true" />
    <type id="0xED" description="Action ID"                name="action_id"         size="1"  discrete="true" />
    <type id="0xEF" description="Transaction ID"           name="trans_id"          size="4"  discrete="true" />
    <type id="0xF0" description="Node ID"                  name="node_id"           size="8"  discrete="true" />
    <type id="0xF1" description="Vendor ID"                name="vendor_id"         size="2"  discrete="true" />
    <type id="0xF2" description="Device Type ID"           name="devtype_id"        size="4"  discrete="true" />
    <type id="0xF3" description="Fabric ID"                name="fabric_id"         size="8"  discrete="true" />
    <type id="0xF4" description="Group ID"                 name="group_id"          size="2"  discrete="true" />
    <type id="0xF5" description="Status Code"              name="status"            size="2"  discrete="true" />
    <type id="0xF6" description="Data Version"             name="data_ver"          size="4"  discrete="true" />
    <type id="0xF7" description="Event Number"             name="event_no"          size="8"  discrete="true" />
    <type id="0xF8" description="Endpoint Number"          name="endpoint_no"       size="2"  discrete="true" />
    <type id="0xF9" description="Fabric Index"             name="fabric_idx"        size="1"  discrete="true" />
```
plus as I said at least `temp-diff` and `temperature`, which as I said are not using named types right now, probably because those don't work for the signed case...

Reply:
I have added the current exceptions into the loader. For future data types which make it into the types xml, I have updated the loader to account for `signed=true` in the xml. So when defining the signed variables just add that. That will take precedence and load the sign right for these data types